docs/ref/contrib/admin/filters.txt

```diff
@@ -0,0 +1,181 @@
+                return queryset.filter(birthday__gte=date(1990, 1, 1),
+                                        birthday__lte=date(1999, 12, 31))
+
+    class PersonAdmin(admin.ModelAdmin):
+        list_filter = (DecadeBornListFilter,)
+
+.. note::
+
+    As a convenience, the ``HttpRequest`` object is passed to the
+    ``lookups`` and ``queryset`` methods, for example::
+
+        class AuthDecadeBornListFilter(DecadeBornListFilter):
+
+            def lookups(self, request, model_admin):
+                if request.user.is_superuser:
+                    return super().lookups(request, model_admin)
+
+            def queryset(self, request, queryset):
+                if request.user.is_superuser:
+                    return super().queryset(request, queryset)
+
+    Also as a convenience, the ``ModelAdmin`` object is passed to
+    the ``lookups`` method, for example if you want to base the
+    lookups on the available data::
+
+        class AdvancedDecadeBornListFilter(DecadeBornListFilter):
+
+            def lookups(self, request, model_admin):
+                """
+                Only show the lookups if there actually is
+                anyone born in the corresponding decades.
+                """
+                qs = model_admin.get_queryset(request)
+                if qs.filter(birthday__gte=date(1980, 1, 1),
+                            birthday__lte=date(1989, 12, 31)).exists():
+                    yield ('80s', _('in the eighties'))
+                if qs.filter(birthday__gte=date(1990, 1, 1),
+                            birthday__lte=date(1999, 12, 31)).exists():
+                    yield ('90s', _('in the nineties'))
+
+Using ``django.contrib.admin.FieldListFilter``
+----------------------------------------------
+
+Specify list_filter using a tuple, where the first element is a field name
+and the second element is a class inheriting from
+``django.contrib.admin.FieldListFilter``, for example::
+
+        class PersonAdmin(admin.ModelAdmin):
+            list_filter = (
+                ('is_staff', admin.BooleanFieldListFilter),
+            )
+
+You can limit the choices of a related model to the objects involved in
+that relation using ``RelatedOnlyFieldListFilter``::
+
+        class BookAdmin(admin.ModelAdmin):
+            list_filter = (
+                ('author', admin.RelatedOnlyFieldListFilter),
+            )
+
+Assuming ``author`` is a ``ForeignKey`` to a ``User`` model, this will
+limit the ``list_filter`` choices to the users who have written a book
+instead of listing all users.
+
+You can filter empty values using ``EmptyFieldListFilter``, which can
+filter on both empty strings and nulls, depending on what the field
+allows to store::
+
+        class BookAdmin(admin.ModelAdmin):
+            list_filter = (
+                ('title', admin.EmptyFieldListFilter),
+            )
+
+.. note::
+
+    The ``FieldListFilter`` API is considered internal and might be
+    changed.
```

Comment:
Just to get @carltongibson's eyes on it; I know it's just moving what was previously there:

This line has been in place since 2012, despite there being prior examples of using a subclass (e.g. `BooleanFieldListFilter`) previously.
The API itself might well be considered internal, but it's been promoted in the docs here for nearly a decade, so people will be using it, so it would probably be subject to the usual  deprecation path on API change anyway. Also the `FieldListFilter` has only been touched once since 2012, to fix [ticket 28202](https://code.djangoproject.com/ticket/28202).

What I'm saying is either as part of this PR, or a subsequent one, this scary warning could probably be considered for removal?

Reply:
Thanks @kezabelle — Yes. Going over all this now, we should probably document this fully, with the available subclasses too.

Reply:
@ssb10 A third possible PR there if you want it once we've got this in. 😄 

(Windows/Sphinx set-up review being the second one.)

Reply:
Yes sure. I will definitely give it a shot. 🙂